### ISSUE TITLE
[NFR]: beforeBind and afterBind for Form

### ISSUE THREAD

**Issue by diegosardina** [phalcon/cphalcon#14598]
Similarly to beforeValidation and afterValidation, please add beforeBind and afterBind

**Comment by Jeckerson** [comment]
Could you describe better with some example?

**Comment by diegosardina** [comment]
Symfony for example ([here](https://symfony.com/doc/current/form/dynamic_form_modification.html)) has pre_set_data and post_set_data events (in old symfony versions they were called pre_bind and post_bind events).

They are useful to modify the form layout based on the underlying model or class data, for example showing or hiding a certain field based on the value of another one.

This is better and more flexible than the various ifs in the form class such in the second example [here](https://docs.phalcon.io/4.0/en/forms#initialization).

**Comment by niden** [comment]
I am not sure where the bind points should be though for the events. 

For instance when the form is instantiated, one can pass an entity object and that will set the values in the relevant fields. Additionally when calling `isValid` with data and an optional object, the data is actually checked there. Finally the most obvious place would be `bind` which is called from `isValid` but not from the constructor.

I believe we need to rethink the way forms work and make it more flexible. There is heavy dependency on the Tag component which doesn't help.

Thoughts?

**Comment by diegosardina** [comment]
> I believe we need to rethink the way forms work and make it more flexible.

I agree, the Form class should centralize all the mapping around a unique setData() or whatever name it will be. This must be took in consideration before implementing embedded forms #705 and form collections #11579 that needs a more laborious mapping strategy.

**Comment by niden** [comment]
We can definitely take a stab at this during v4. I would like to see what the community thinks

**Comment by niden** [comment]
Closing this: Please vote for this feature here: https://github.com/phalcon/cphalcon/issues/14608

**Comment by niden** [comment]
Resolved in https://github.com/phalcon/cphalcon/pull/17082

Thank you @diegosardina
